src/main/java/dk/kb/labsapi/metadataFormats/IllustrationMetadata.java

```diff
@@ -24,48 +24,50 @@
  *     <li>x: height of illustration</li>
  * </ul>
  */
-public class IllustrationMetadata {
+public class IllustrationMetadata extends BasicMetadata {
     private static final Logger log = LoggerFactory.getLogger(IllustrationMetadata.class);
-    private String pageUUID;
-    private double pageWidth;
-    private double pageHeight;
     private String id;
     private double x;
     private double y;
     private double w;
     private double h;
-    static final Pattern illustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*),doms_aviser_page:uuid:(\\S*),(\\d*),(\\d*)");
+    static final Pattern oldIllustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*),doms_aviser_page:uuid:(\\S*),(\\d*),(\\d*)");
+    static final Pattern singleIllustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*)");
 
-    public IllustrationMetadata(){
-    }
 
-    // Setter
-    public void setData(String input) {
-        Matcher m = illustrationPattern.matcher(input);
+    public IllustrationMetadata(String input){
+        Matcher m = oldIllustrationPattern.matcher(input);
         if (m.matches()) {
             this.id = m.group(1);
             this.x = Double.parseDouble(m.group(2));
             this.y = Double.parseDouble(m.group(3));
             this.w = Double.parseDouble(m.group(4));
             this.h = Double.parseDouble(m.group(5));
             this.pageUUID = m.group(6);
-            this.pageWidth = convertPixelsToInch1200(Integer.parseInt(m.group(7)));
-            this.pageHeight = convertPixelsToInch1200(Integer.parseInt(m.group(8)));
+            this.pageWidth = (long) convertPixelsToInch1200(Integer.parseInt(m.group(7)));
+            this.pageHeight = (long) convertPixelsToInch1200(Integer.parseInt(m.group(8)));
         } else {
             log.error("Regex matching failed. Please check that input and pattern aligns.");
         }
     }
 
-    // Getters
-    public String getPageUUID(){
-        return pageUUID;
-    }
-    public double getPageWidth() {
-        return pageWidth;
-    }
-    public double getPageHeight() {
-        return pageHeight;
+    public IllustrationMetadata(String illustrationString, String pageUUID, long pageWidth, long pageHeight){
```

Comment:
Please add JavaDoc, such as where the `illustrationString`comes from and maybe with an example of such a String?

Reply:
Teeny tiny JavaDoc hint: For code snippets, data samples and similar, the notation `{@code id=ART88-1_SUB,x=2364,y=4484,w=652,h=100}` is preferable.